src/context/RequestClientContext.tsx

```diff
@@ -1,16 +1,98 @@
-import { createContext, FC } from "react";
+import { createContext, FC, useReducer, useCallback } from "react";
 import { oracle } from "@uma/sdk";
 import config from "constants/oracleConfig";
 
 const client = oracle.client.factory(config, () => undefined);
 
-export const RequestClientContext = createContext<oracle.client.Client>(client);
+interface ContextState {
+  client: oracle.client.Client;
+  user: Partial<oracle.types.state.User> | null;
+  request: Partial<oracle.types.state.Inputs> | undefined;
+  setUser: (payload: oracle.types.state.User) => void;
+  setActiveRequest: (payload: oracle.types.state.Inputs) => void;
+}
+export const RequestClientContext = createContext<ContextState>(
+  {} as ContextState
+);
 
 RequestClientContext.displayName = "RequestClientContext";
 
+const SET_USER = "set_user";
+const SET_ACTIVE_REQUEST = "set_active_request";
+const UPDATE_CLIENT_STATE = "update_client_state";
+
+type Action =
+  | {
+      type: typeof SET_USER;
+      payload: oracle.types.state.User;
+    }
+  | {
+      type: typeof SET_ACTIVE_REQUEST;
+      payload: oracle.types.state.Inputs;
+    }
+  | {
+      type: typeof UPDATE_CLIENT_STATE;
+    };
+
+interface ClientState {
+  user: Partial<oracle.types.state.User> | null;
+  request: Partial<oracle.types.state.Inputs> | undefined;
+}
+
+function reducer(state: ClientState, action: Action): ClientState {
+  switch (action.type) {
+    case SET_USER: {
+      client.setUser(
+        action.payload.address,
+        action.payload.chainId,
+        action.payload.signer
+      );
+      client.update.all().catch((err: any) => undefined);
```

Comment:
Sorry if this q has been answered before, but what does this do exactly? 
If its a side effect that needs to run when the user is set, I think ideally it should have its own useEffect and we should keep the reducer pure (but we can keep as is for now).

If its something that needs to run before we insert the user into state, then I think we should rethink this pattern, as the user will be inserted into state before this async side effect has finished running.  One solution would be to move this promise into its own useEffect and call a setUser in the .then of the promise.

Reply:
@daywiss can probably explain it better, but the client is a bit immature, so it doesn't update state until I call this. but it just aligns the FE + client.